src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -513,44 +510,94 @@ public void Initialize(ITelemetry telemetry)
             }
 
             telemetry.Context.Initialize(this.Context, instrumentationKey);
-            foreach (ITelemetryInitializer initializer in this.configuration.TelemetryInitializers)
+
+            ISupportAdvancedSampling telemetryWithSampling = telemetry as ISupportAdvancedSampling;
+
+            // Telemetry can be already sampled out if that decision was made before calling Track()
+            bool sampledOut = telemetryWithSampling?.IsProactivelySampledOut ?? false;
+
+            // If telemetry was not sampled out, we need to check its sampling score after we learn its Operation Id
+            bool shoudlCheckSamplingScore = !sampledOut;
+
+            if (!sampledOut)
             {
-                try
+                if (telemetryWithSampling?.SupportsProactiveSampling ?? false)
```

Comment:
hm, it seems it is already checked with first `if (!sampledOut)`

Reply:
`if (!sampledOut)` checks for case when telemetry was already sampled out and we do not need to run initializers.
`if (supportsProactiveSampling)` checks whether current telemetry type supports proactive sampling and we may skip initializers after we know its operation ID or we need to run all initializers.